components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -401,36 +401,37 @@ SUBROUTINE p3_main_pre_main_loop(kts, kte, kbot, ktop, kdir, log_predictNc, dt,
        inv_rho(k) = 1._rtype/rho(k)
        qvs(k)     = qv_sat(t(k),pres(k),0)
        qvi(k)     = qv_sat(t(k),pres(k),1)
-
-       sup(k)     = qv(k)/qvs(k)-1._rtype
+                        
        supi(k)    = qv(k)/qvi(k)-1._rtype
 
        rhofacr(k) = bfb_pow(rhosur*inv_rho(k), 0.54_rtype)
        rhofaci(k) = bfb_pow(rhosui*inv_rho(k), 0.54_rtype)
        dum        = 1.496e-6_rtype * bfb_pow(t(k), 1.5_rtype) / (t(k)+120._rtype)  ! this is mu
        acn(k)     = g*rhow/(18._rtype*dum)  ! 'a' parameter for droplet fallspeed (Stokes' law)
 
-       !specify cloud droplet number (for 1-moment version)
-       if (.not.(log_predictNc)) then
-          nc(k) = nccnst*inv_rho(k)
-       endif
 
-       if ((t(k).lt.zerodegc .and. supi(k).ge.-0.05_rtype) .or.                              &
-            (t(k).ge.zerodegc .and. sup(k).ge.-0.05_rtype )) log_nucleationPossible = .true.
+       if (t(k).lt.zerodegc .and. supi(k).ge.-0.05_rtype) log_nucleationPossible = .true.
 
        !--- apply mass clipping if dry and mass is sufficiently small
        !    (implying all mass is expected to evaporate/sublimate in one time step)
 
-       if (qc(k).lt.qsmall .or. (qc(k).lt.1.e-8_rtype .and. sup(k).lt.-0.1_rtype)) then
+       if (qc(k).lt.qsmall) then
           qv(k) = qv(k) + qc(k)
           th(k) = th(k) - exner(k)*qc(k)*xxlv(k)*inv_cp
           qc(k) = 0._rtype
           nc(k) = 0._rtype
        else
           log_hydrometeorsPresent = .true.    ! updated further down
+               
+          !Droplet activation OR specify cloud droplet number (for 1-moment version)
```

Comment:
The comment just above is a bit confusing and needs something to point out that it is actually an important line in the code. Maybe "!--- Apply droplet activation here (before other microphysical processes) for consistency with qc increase by saturation adjustment already applied in macrophysics. If prescribed drop number is used, this is also a good place to prescribe that value."